kubelink/pkg/service/helmApplicationService/utils.go

```diff
@@ -33,3 +47,156 @@ const (
 func IsReleaseNotFoundInCacheError(err error) bool {
 	return errors.Is(err, k8sInformer.ErrorCacheMissReleaseNotFound)
 }
+
+// UpdateChartLock updates the Chart.lock file based on its contents.
+func UpdateChartLock(lockFilePath string, helmRelease *release.Release) error {
+	lockFileData, err := os.ReadFile(lockFilePath)
+	if err != nil {
+		return fmt.Errorf("error reading Chart.lock file at %s: %w", lockFilePath, err)
```

Comment:
1. seems like a business logic. should be binded with our interface/ implemention. 
2. use suggar logger instead.
3. On err, we are not returning err. Is it intentional ? 
if err != nil  DO THE BELOW
```
if err := yaml.Unmarshal(lockFileData, helmRelease.Chart.Lock); err != nil {
		return fmt.Errorf("error unmarshalling Chart.lock file at %s: %w", lockFilePath, err)
	}
```
Else initialise. use proper if else convention for better readability.